### ISSUE TITLE
Poincare - Link Prediction train set should not exclude links with leaves

### ISSUE THREAD

**Issue by octavian-ganea** [RaRe-Technologies/gensim#1857]
#### Description
Links containing root/leaves should not be excluded from the train set of the Link Prediction task in the Poincare embeddings code.

#### Steps/Code/Corpus to Reproduce
In the Poincare embeddings paper it is stated that links containing the root or a leaf node should be excluded from the validation and test set. However, in the Gensim code they are excluded also from the train set. This cannot be useful since there will be no knowledge at all about leaves at both train and test time, making the task much harder and potentially explaining the worse results obtained so far.

The function that generates the train file is `train_test_split(data_file, test_ratio=0.1)` from
https://github.com/RaRe-Technologies/gensim/blob/develop/docs/notebooks/Poincare%20Evaluation.ipynb

@jayantj can you please elaborate?

**Comment by octavian-ganea** [comment]
Sorry, your implementation is correct.
